packages/web-components/src/components/inline-loading/inline-loading.ts

```diff
@@ -28,24 +28,66 @@ export { INLINE_LOADING_STATE };
 @customElement(`${prefix}-inline-loading`)
 class CDSInlineLoading extends LitElement {
   /**
-   * The assistive text for the spinner icon.
+   * @deprecated The 'assistive-text' property will be deprecated in the next major release. Please use `icon-description` instead.
    */
   @property({ attribute: 'assistive-text' })
-  assistiveText = 'Loading';
+  get assistiveText() {
+    return this.iconDescription;
+  }
+  set assistiveText(value) {
+    this.iconDescription = value;
+  }
+  /**
+   * The assistive text for the spinner icon.
+   */
+  @property({ attribute: 'icon-description' })
+  iconDescription = 'Loading';
+
+  /**
+   * Provide a delay for the setTimeout for success
+   */
+  @property({ attribute: 'success-delay' })
+  successDelay = 1500;
 
   /**
    * @returns The template for the status icon.
    */
   private _renderIcon() {
-    const { assistiveText, status } = this;
+    const { iconDescription, status } = this;
     if (status === INLINE_LOADING_STATE.ERROR) {
       return ErrorFilled16({
         class: `${prefix}--inline-loading--error`,
+        children: html`<title>${iconDescription}</title>`,
       });
     }
+    const init = {
+      bubbles: true,
+      cancelable: true,
+      composed: true,
+    };
+
     if (status === INLINE_LOADING_STATE.FINISHED) {
+      if (
+        this.dispatchEvent(
+          new CustomEvent(
+            (this.constructor as typeof CDSInlineLoading).eventOnSuccess,
+            init
+          )
+        )
+      ) {
+        setTimeout(() => {
+          this.dispatchEvent(
+            new CustomEvent(
+              (this.constructor as typeof CDSInlineLoading).eventOnSuccess,
+              init
+            )
+          );
+        }, this.successDelay);
+      }
```

Comment:
```suggestion
    setTimeout(() => {
          this.dispatchEvent(
            new CustomEvent(
              (this.constructor as typeof CDSInlineLoading).eventOnSuccess,
              init
            )
          );
        }, this.successDelay);
```

the on-success event is actually getting triggered at the `if` statement, so with the timeout  the event is triggered twice